src/topology/builder.rs

```diff
@@ -402,8 +402,20 @@ impl<'a> Builder<'a> {
         {
             debug!(component = %key, "Building new transform.");
 
-            let input_definitions =
-                schema::input_definitions(&transform.inputs, self.config, &mut definition_cache);
+            let input_definitions = match schema::input_definitions(
+                &transform.inputs,
+                self.config,
+                enrichment_tables.clone(),
+                &mut definition_cache,
+            ) {
+                Ok(definitions) => definitions,
+                Err(_) => {
+                    // We have recieved an error whilst retrieving the definitions,
+                    // there is no point in continuing.
+
+                    return;
+                }
```

Comment:
Not being very familiar with the topology code, is this new behavior something we could unit test? (Or is that covered by tests you added in the PR in other areas of the code that exercise this code path)?

Reply:
I have added another test.